### DISCUSSION TITLE
Production ready?

### DISCUSSION THREAD

**Discussion by bmjhversteeg** [maoosi/prisma-appsync#25] [Q&A] [answered]
First of all, thanks for this amazing generator. Had some problems setting up the Lambda, but now it is working and performs absolutely brilliant. This generator was exactly what I was looking for, because the resolvers from Appsync are very limiting, but the auth options and graphql subscriptions in combination with Amplify are nice.

My question: You state that this generator is not for production envs yet. Could you elaborate on when it is considered "Production-ready"? Do you foresee breaking changes? What is the schedule for working towards the stable release?

Thanks.

**Comment by maoosi** [comment, marked as answer]
@bmjhversteeg Thanks for your kind words, always good to hear about people happily using the Generator!

Two main things need to be addressed before I would consider Prisma-AppSync officially stable:

1. Refactoring and improving tests to cover more scenarios and edge cases.
2. There is a known issue with [fine-grained access control](https://prisma-appsync.vercel.app/guides/securing-api.html#%F0%9F%91%89-fine-grained-access-control). To sum up, the current implementation is not efficiently protecting nested fields, which could lead to some security issues.

I am currently working on a new version of the Client API (the part of the Generator used inside the Lambda function). Breaking changes will come with this new version, mainly impacting people that are extensively using custom resolvers and fine-grained access control. If you are using Prisma-AppSync in a more basic way, then the migration should be easy.

More info and example code of what’s coming next can be found on that issue: https://github.com/maoosi/prisma-appsync/issues/18

The work on this has already started (probably 50% done), though I'm not able to commit to a proper schedule at this stage, as I am only working on it sporadically. I wouldn’t expect anything before 1-2 months from now.

I’ll update this issue as soon as I have a clearer view of the timeline.
